### DISCUSSION TITLE
PMI  association with feature surface

### DISCUSSION THREAD

**Discussion by ly155331445** [Open-Cascade-SAS/OCCT#75] [Q&A]
May I ask if the latest version of OCC supports outputting STP 242 format with PMI correlation?

**Comment by dpasukhi** [comment]
@ly155331445 Hello
Could you please a little extend your request?
Is it a question? What PMI correlation is?
`Outputting ` is it a Export or Import?

**Reply by ly155331445** [reply]
when OCC export a Step242 file, can the relationship between PMI and surfaces be recorded in the Step242 file?  
for example , A length dimension annotation, we neet it  record the two faces related with it .
We want to use this  Step242 file  in  NX  .

**Reply by dpasukhi** [reply]
Yes, it is possible. Moreover, any PMI, in stp must to have annotation.
Any method in `XCAFDoc_DimTolTool `asks for connection to geometry.
Please check what done in https://github.com/Open-Cascade-SAS/OCCT/blob/495a6a642dbfcebd866a6d6f93b4d9c4efa5b415/src/STEPCAFControl/STEPCAFControl_Reader.cxx#L2757-L3002
Just as a sample for the adding PMI

**Reply by ly155331445** [reply]
Thanks . And what methods can I use to verify? Maybe use freeCAD to export a Step242 file ?

**Reply by dpasukhi** [reply]
First of all you need fill up the PMIs labels using XCAFDoc_DimTolTool and DimensionObjects.
Export can be done by pure OCCT.

**Reply by wwll** [reply]
> Yes, it is possible. Moreover, any PMI, in stp must to have annotation. Any method in `XCAFDoc_DimTolTool `asks for connection to geometry. Please check what done in
> 
> https://github.com/Open-Cascade-SAS/OCCT/blob/495a6a642dbfcebd866a6d6f93b4d9c4efa5b415/src/STEPCAFControl/STEPCAFControl_Reader.cxx#L2757-L3002
> 
> 
> Just as a sample for the adding PMI

I have similiar problem .  I would like to do the following with PMI: obtain the PMI, and access the STEP entities that contain the physical geometries that the PMI is referencing. How can I achieve achieve my purpose ?

**Reply by wwll** [reply]
I have similiar problem . I would like to do the following with PMI: obtain the PMI, and access the STEP entities that contain the physical geometries that the PMI is referencing. How can I achieve achieve my purpose ?

**Reply by dpasukhi** [reply]
@ly155331445  what you want to varify? The export process?
The exported files can be validated in the https://www.nist.gov/services-resources/software/step-file-analyzer-and-viewer
It is a NIST application, based on OCCT.
_
@wwll you request moved to the https://github.com/Open-Cascade-SAS/OCCT/discussions/92
